cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -160,6 +160,9 @@ func Run(s *options.CMServer) error {
 	if err != nil {
 		return err
 	}
+	// it is more safe to use hostname plus pod name
+	// see https://github.com/kubernetes/kubernetes/issues/22775
+	id = id + "-" + os.Getenv("POD_NAME")
```

Comment:
I see the `port` is overridable via command line, so if containerPort is the way, maybe we should not allow this.